### ISSUE TITLE
Command single device

### ISSUE THREAD

**Issue by danieledge** [KraigM/homebridge-harmonyhub#9]
Hi,

The plugin seems to work really well for activities. However, I'd like to be able to toggle a single device e.g. I have an IR based mains plug. I can wrap this in an activity, but this then powers off other devices as it switches to activities.

Can the functionality to toggle the power of a single device be added?

Thanks,

D

**Comment by KraigM** [comment]
Yeah this is something I def want to do (or anyone else is free to jump in and work on it).  Hard part is that HomeKit is limited to 100 devices right now.  Which is good I suppose, as it forces people to truly treat each device as a single device.

**Comment by KraigM** [comment]
When we do, we probably should make configs so that it is optional (i.e. whitelists, blacklists, and/or opt in/out)

**Comment by danieledge** [comment]
Looking back at the original implementation/code of the plugin it looks like this contained code that retrieved the device list. Do you know whether this actually worked ?

Thanks

**Comment by KraigM** [comment]
Looking at the [client api](https://github.com/swissmanu/harmonyhubjs-client/blob/master/lib/harmonyclient.js), it looks like he is using a **getAvailableCommands** method and accessing the _activities_ property to get the activities.  Not sure what else gets returned (would have to debug into it), but it might contain things that we could use (like a _devices_ property hopefully).

**Comment by leoneleone** [comment]
The most useful command to add as a starter would be Volume control. Perhaps that can be implemented as an independent device in HomeKit, "Harmony Volume Control" for example

**Comment by kimme1024** [comment]
Sorry to bump this thread after 3 years but I can't find any solution on this matter.
I have a IR-controllable fireplace that I can use with my Harmony Hub.
I have Homebridge running but you can only control the activities. 
I want to have an option to turn on my fireplace trough Homebridge when I'm on my way home but I don't want it to turn of when I, let's say, start the "Watch Netflix" activity when I'm home.

Is there any solution for this in the meantime?

Thanks in advance!

**Comment by Makk1000** [comment]
Add the fireplace to turn on as well with the Netflix. It’s what I did with my Dyson fan

**Comment by kimme1024** [comment]
That's not a solution.
I want to be able to start the fireplace on my way home, so why start netflix together with it.
Also, when you switch from let's say netflix to radio the fireplace will turn off, or when you shut down the fireplace, netflix turns off... 

Long story short: not an option :)
